query-graphs/src/tree-rendering.ts

```diff
@@ -581,11 +582,52 @@ export function drawQueryTree(target: HTMLElement, treeData: TreeDescription) {
         zoomBehavior.translateTo(baseSvg, x, y);
     }
 
+    // Scale the graph so that it can fit nicely on the screen
+    function fitGraphScale() {
+        // Get the bounding box of the main SVG element, we are interested in the width and height
+        const bounds = baseSvg.node().getBBox();
+        // Get the size of the container of the main SVG element
+        const parent = baseSvg.node().parentElement;
+        const fullWidth = parent.clientWidth;
+        const fullHeight = parent.clientHeight;
+        // Let's find the scale factor
+        // Thinking of only the X dimension, we need to make the width match the fullWidth, Equation:
+        //   width * xfactor = fullWidth
+        // Solving for `xfactor` we have: xfactor = fullWidth/width
+        const xfactor: number = fullWidth / bounds.width;
+        // Similarly, we would find out that: yfactor = fullHeight/height;
+        const yfactor: number = fullHeight / bounds.height;
+        // Most likely, the X and Y factor are different. We use the minimum of them to avoid cropping
+        let scaleFactor: number = Math.min(xfactor, yfactor);
+        // Add some padding so that the graph is not touching the edges
+        const paddingPercent = 0.9;
+        scaleFactor = scaleFactor * paddingPercent;
+        console.log("Scale factor", scaleFactor);
+        zoomBehavior.scaleBy(baseSvg, scaleFactor);
+    }
+
+    // Center the graph (without scaling)
+    function centerGraph() {
+        // Find the Bounding box center, then translate to it
+        // In other words, put the bbox center in the center of the screen
+        const bbox = svgGroup.node().getBBox();
+        console.log("BBox", bbox);
+        const cx = bbox.x + bbox.width / 2;
+        const cy = bbox.y + bbox.height / 2;
+        console.log("BBox center: ", cx, cy);
+        zoomBehavior.translateTo(baseSvg, cx, cy);
+    }
+
     // Scale for readability by a fixed amount due to problematic .getBBox() above
     zoomBehavior.scaleBy(baseSvg, 1.5);
 
     orientRoot();
 
+    const infoCard = d3selection
+        .select(target)
+        .append("div")
+        .attr("id", "info-card")
```

Comment:
In this case the id seems to be completely unused...

Reply:
removing...